### PR TITLE
[v6r21] Do not follow links when getting files list or size in directory

### PR DESCRIPTION
**Pull request by fstagni**
This was creating infinite loops.

BEGINRELEASENOTES

*Core
CHANGE: Do not follow links when getting files list or size in directory via getGlobbedFiles and getGlobbedTotalSize

ENDRELEASENOTES

### CODE REVIEW

**Comment by chaen** [comment]
Hum, I would rather have it an option, keeping the current behaviour as default. This can have nasty side effect.

File: Core/Utilities/File.py [hunk 1]
```diff
@@ -179,7 +176,7 @@ def getGlobbedTotalSize(files):
   else:
     for path in glob.glob(files):
       if os.path.isdir(path):
-        for content in os.listdir(path):
+        for content in os.listdir(path) and not os.path.islink(path):
```
Comment (andresailer) [inline Core/Utilities/File.py]: The `and not...` has to be moved to the if statement in the line above.

Comment (fstagni) [reply Core/Utilities/File.py]: ah yes thanks

**Comment by fstagni** [comment]
We have already a nasty side effect, that is exemplified by the following:

```
 fstagni  ⓔ DIRAC  ~/aaaRumenta/toDelete   ll
total 8
drwxrwxr-x 2 fstagni fstagni 4096 Feb  7 11:27 .
drwxr-xr-x 4 fstagni fstagni 4096 Feb  7 11:27 ..
-rw-rw-r-- 1 fstagni fstagni    0 Feb  7 11:27 pippo
lrwxrwxrwx 1 fstagni fstagni    1 Feb  7 11:27 pluto -> .
 fstagni  ⓔ DIRAC  ~/aaaRumenta/toDelete   ipy
Python 2.7.13 |Anaconda, Inc.| (default, Sep 30 2017, 18:12:43) 
Type "copyright", "credits" or "license" for more information.

IPython 5.3.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from DIRAC.Core.Utilities.File import getGlobbedFiles

In [2]: getGlobbedFiles('pluto')
Out[2]: 
['pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pluto/pippo',
 'pluto/pluto/pluto/pippo',
 'pluto/pluto/pippo',
 'pluto/pippo']
```

**Comment by chaen** [comment]
Can you at least print a warning or something, telling that the folder will not be traversed ?

**Comment by fstagni** [comment]
I think what's now is the correct behavior, so I am not sure it's worth a "WARNING". What do others think?

**Pull request merged**